Doc/whatsnew/3.12.rst

```diff
@@ -62,91 +62,171 @@ Summary -- Release highlights
+as :term:`f-strings <f-string>` have had many limitations removed
+and 'Did you mean ...' suggestions continue to improve.
+The new :ref:`type parameter syntax <whatsnew312-pep695>`
+and :keyword:`type` statement improve ergonomics for using :term:`generic types
+<generic type>` and :term:`type aliases <type alias>` with static type checkers.
+
+This article doesn't attempt to provide a complete specification of all new features,
+but instead tries to give a convenient overview.
+For full details, you should refer to the documentation,
+such as the  :ref:`Library Reference <library-index>`
+and :ref:`Language Reference <reference-index>`.
+If you want to understand the complete implementation and design rationale for a change,
+refer to the PEP for a particular new feature;
+but note that PEPs usually are not kept up-to-date
+once a feature has been fully implemented.
+
+--------------
 
 .. PEP-sized items next.
 
+New syntax features:
+
+* :ref:`PEP 695 <whatsnew312-pep695>`, type parameter syntax and the :keyword:`type` statement
+
 New grammar features:
 
-* :ref:`whatsnew312-pep701`
+* :ref:`PEP 701 <whatsnew312-pep701>`, :term:`f-strings <f-string>` in the grammar
 
 Interpreter improvements:
 
-* :ref:`whatsnew312-pep684`
+* :ref:`PEP 684 <whatsnew312-pep684>`, a unique per-interpreter :term:`GIL
+  <global interpreter lock>`
+* :ref:`PEP 669 <whatsnew312-pep669>`, low impact monitoring
+* `Improved 'Did you mean ...' suggestions <improved error messages_>`_
+  for :exc:`NameError`, :exc:`ImportError`, and :exc:`SyntaxError` exceptions
 
-* :ref:`whatsnew312-pep669`
+Python data model improvements:
 
-New typing features:
+* :ref:`PEP 688 <whatsnew312-pep688>`, using the :ref:`buffer protocol
+  <bufferobjects>` from Python
+
+Significant improvements in the standard library:
+
+* The :class:`pathlib.Path` class now supports subclassing
+* The :mod:`os` module received several improvements for Windows support
+* A :ref:`command-line interface <sqlite3-cli>` has been added to the
+  :mod:`sqlite3` module
+* :func:`isinstance` checks against :func:`runtime-checkable protocols
+  <typing.runtime_checkable>` enjoy a speed up of between two and 20 times
+* A :ref:`command-line interface <uuid-cli>` has been added to the
+  :mod:`uuid` module
 
-* :ref:`whatsnew312-pep688`
+Security improvements:
 
-* :ref:`whatsnew312-pep692`
+* Replace the builtin :mod:`hashlib` implementations of
+  SHA1, SHA3, SHA2-384, SHA2-512, and MD5 with formally verified code from the
+  `HACL* <https://github.com/hacl-star/hacl-star/>`__ project.
 
-* :ref:`whatsnew312-pep695`
+C API improvements:
+
+* :ref:`PEP 697 <whatsnew312-pep697>`, unstable C API tier
+* :ref:`PEP 683 <whatsnew312-pep683>`, immortal objects
+
+CPython implementation improvements:
+
+* :ref:`PEP 709 <whatsnew312-pep709>`, comprehension inlining
+* :ref:`CPython support <perf_profiling>` for the Linux ``perf`` profiler
+* Implement stack overflow protection on supported platforms
```

Comment:
cc: @markshannon -- I thought it was worth noting the stack overflow change (#91079) given there've been a few comments about the change to the recursion limit & C stack, would welcome views.

Also cc @mdboom @Fidget-Spinner @iritkatriel (unsure who else) -- are there any other implementation improvements e.g. from Faster CPython that we should draw out in the release summary?

Reply:
@iritkatriel -- should we note the 3.12 changes to the compiler/assembler/CFG (e.g. #87092) here under 'CPython implementation improvements'?